packages/slotted-label-mixin/slotted-label-mixin.ts

```diff
@@ -0,0 +1,42 @@
+import { LitElement, PropertyValues } from 'lit-element';
+import { Constructor } from '@vaadin/mixin-utils';
+
+const isEmpty = (nodes: Node[]) => {
+  return (
+    nodes.length === 0 ||
+    (nodes.length === 1 &&
+      nodes[0].nodeType === Node.TEXT_NODE &&
+      (nodes[0].textContent || '').trim() === '')
+  );
+};
+
+export const SlottedLabelMixin = <T extends Constructor<LitElement>>(base: T): T => {
+  class SlottedLabel extends base {
+    private _defaultPart = '';
+
+    protected firstUpdated(props: PropertyValues) {
+      super.firstUpdated(props);
+
+      const label = this.renderRoot.querySelector('[part~="label"]');
+      if (label) {
+        this._defaultPart = label.getAttribute('part') as string;
+        const slot = label.firstElementChild;
+        if (slot instanceof HTMLSlotElement) {
+          slot.addEventListener('slotchange', () => this._updateLabelPart(label, slot));
+          this._updateLabelPart(label, slot);
+        }
+      }
+    }
+
+    private _updateLabelPart(label: Element, slot: HTMLSlotElement) {
+      const part = label.getAttribute('part');
+      const newPart = this._defaultPart + (isEmpty(slot.assignedNodes()) ? ' label-empty' : '');
+
+      if (part !== newPart) {
+        label.setAttribute('part', newPart);
```

Comment:
Shouldn't the `part` attributes always be static? Emptyness seems more like a _state_ so something like `<div part="label" empty>` might be more logical than `<div part="label label-empty">`.

Reply:
Yes that's how it previously was, but in that case users would not be able to use `::part()` selector and we don't have `:state` yet.

I'nm ok to revert it for now.

Reply:
Discussed in person and decided to use `label-empty` host attribute instead.

Reply:
Changed.